test/blackbox/common/DDSBlackboxTestsStatistics.cpp

```diff
@@ -166,9 +156,52 @@ TEST(DDSStatistics, simple_statistics_datareaders)
     auto r_statistics_participant = statistics::dds::DomainParticipant::narrow(r_participant);
     ASSERT_NE(nullptr, r_statistics_participant);
 
-    // Enable DataReader related statistics
-    auto ack_stats_reader = enable_statistics(r_statistics_participant, r_subscriber, statistics::ACKNACK_COUNT_TOPIC);
-    ASSERT_NE(nullptr, ack_stats_reader);
+    // TODO: some topics get stuck in infinite loop in an error:
+    // [SUBSCRIBER Error] Change not found on this key, something is wrong -> Function remove_change_sub
+    // These topics are commented in test params
+    // TODO: some topics could be used in both participants, but they lead to the same error
+
+    // Create parameters to iterate over every Statistics kind
+    // The test is separated between the statistics retrieved by a DataWriter or a DataReader
+    std::vector<std::tuple<std::string, std::string, std::size_t>> writer_statistics_kinds = {
+        {"DATA_COUNT_TOPIC",                statistics::DATA_COUNT_TOPIC,               num_samples},
+        {"RTPS_SENT_TOPIC",                 statistics::RTPS_SENT_TOPIC,                num_samples},
+        {"NETWORK_LATENCY_TOPIC",           statistics::NETWORK_LATENCY_TOPIC,          num_samples},
+        {"PUBLICATION_THROUGHPUT_TOPIC",    statistics::PUBLICATION_THROUGHPUT_TOPIC,   num_samples},
+        {"HEARTBEAT_COUNT_TOPIC",           statistics::HEARTBEAT_COUNT_TOPIC,          num_samples},
+        // {"RTPS_SENT_TOPIC",                 statistics::RTPS_SENT_TOPIC,                num_samples},
+        // {"DATA_COUNT_TOPIC",                statistics::DATA_COUNT_TOPIC,               num_samples},
```

Comment:
These lines were already included several lines above and can be removed

Reply:
:+1: